LINQ/Task1/LinqTask.cs

```diff
@@ -0,0 +1,154 @@
+                .Select(x => (x, x.Orders.Select(y => y.OrderDate).Min()));
+        }
+
+        public static IEnumerable<(Customer customer, DateTime dateOfEntry)> Linq5(
+            IEnumerable<Customer> customers
+        )
+        {
+            return customers
+                .Where(x => x.Orders.Count() > 0)
+                .Select(x => (x, x.Orders.Select(y => y.OrderDate).Min()))
+                .OrderBy(x => x.Item2.Year)
+                .ThenBy(x => x.Item2.Month)
+                .ThenByDescending(x => x.x.Orders.Select(x => x.Total).Sum())
+                .ThenBy(x => x.x.CompanyName);
+        }
+
+        public static IEnumerable<Customer> Linq6(IEnumerable<Customer> customers)
+        {
+            return customers.Where(x => Regex.IsMatch(x.PostalCode, @"[\D]")
+            || string.IsNullOrEmpty(x.Region)
+            || !Regex.IsMatch(x.Phone, @"[\(\+?]"));
+        }
+
+        public static IEnumerable<Linq7CategoryGroup> Linq7(IEnumerable<Product> products)
+        {
+            /* example of Linq7result
+
+             category - Beverages
+	            UnitsInStock - 39
+		            price - 18.0000
+		            price - 19.0000
+	            UnitsInStock - 17
+		            price - 18.0000
+		            price - 19.0000
+             */
+
+            var group = products.GroupBy(x => new { x.Category }).Select(y => new
+            {
+                Category = y.Key.Category,
+                UnitInStockGroups = y.GroupBy(z => z.UnitsInStock).Select(z => new
+                {
+                    UnitsInStock = z.Key,
+                    Products = z.OrderBy(v => v.UnitPrice)
+                })
+            });
+
+            var result = group.Select(x => new Linq7CategoryGroup()
+            {
+                Category = x.Category,
+                UnitsInStockGroup = x.UnitInStockGroups.Select(z => new Linq7UnitsInStockGroup()
+                {
+                    UnitsInStock = z.UnitsInStock,
+                    Prices = z.Products.Select(x => x.UnitPrice)
+                })
+            });
+
+            return result;
+        }
+
+        public static IEnumerable<(decimal category, IEnumerable<Product> products)> Linq8(
+        IEnumerable<Product> products,
+        decimal cheap,
+        decimal middle,
+        decimal expensive
+    )
+        {
+            var categories = new decimal[] { cheap, middle, expensive };
+
+            return products.GroupBy(x => categories.FirstOrDefault(y => x.UnitPrice <= y)).Select(x => (x.Key, x.AsEnumerable()));
+        }
+
+        public static IEnumerable<(string city, int averageIncome, int averageIntensity)> Linq9(
+            IEnumerable<Customer> customers
+        )
+        {
+            var customersGroupedByCity = customers.GroupBy(x => x.City);
+
+            var result = customersGroupedByCity.Select(x => (
+            x.Key,
+            (int)Math.Round(x.Average(y => y.Orders.Length == 0 ? 0 : y.Orders.Sum(z => z.Total))),
```

Comment:
my suggestion is to extract such logic in order to increase the readability of the code.